views/apps.ejs

```diff
@@ -38,7 +38,7 @@
       var result = [];
       for(var i = 0; i <= data.length-1; i++){
         console.log("Printing data "+i+" : "+data[i].imageUrl);
-        result.push(data[i].imageUrl);
+        result.push('<ImagePost imageUrl='+data[i].imageUrl+'/>');
```

Comment:
We are pushing image in your code 
```jsx
result.push('<ImagePost imageUrl='+data[i].imageUrl+'/>');
```
But, we should be pushing the react component itself like this 
```jsx
result.push(<ImagePost imageUrl={data[i].imageUrl} />);
```

I am not sure if the thing I am suggesting is possible. Lets try it out and learn react :D